model/vcs/metrics.yaml

```
@@ -0,0 +1,147 @@
+groups:
+  - id: metric.vcs.change.count
+    type: metric
+    metric_name: vcs.change.count
+    brief: 'The number of changes (pull requests/merge requests/changelists) in a repository, categorized by their state (e.g. open or merged)'
+    instrument: updowncounter
+    unit: "{change}"
+    stability: experimental
+    attributes:
+      - ref: vcs.change.state
+        requirement_level: required
+      - ref: vcs.repository.url.full
+        requirement_level: required
+  - id: metric.vcs.change.duration
+    type: metric
+    metric_name: vcs.change.duration
+    brief: 'The time duration a change (pull request/merge request/changelist) has been in a given state.'
+    instrument: gauge
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.change.state
+        requirement_level: required
+  - id: metric.vcs.change.time_to_approval
+    type: metric
+    metric_name: vcs.change.time_to_approval
+    brief: 'The amount of time it took a change (pull request/merge request/changelist) to go from open to approved'
```

nit:

should we clarify if it's a first approval or should it be recorded every time anyone approves a change?

---

It when the state of the change itself changes, which, on the backend, I think would be time to first approval.

---

let's add it to the brie then, e.g. 
```suggestion
    brief: 'The amount of time it took a change (pull request/merge request/changelist) to get the first approval'
```

---

I applied the brief change in fc3096e06ef2b25f2aedbf2cd09150a8273c9f55

---

After discussion in CICD SIG, let's specify it's from creation time.

---

Done in 879f9f1ceaf0f76c5a193ce4423dfd7b334f7b1d.